### PR TITLE
add support for custom operators in postgres

### PR DESCRIPTION
**Pull request by lovasoa**
Fixes https://github.com/sqlparser-rs/sqlparser-rs/issues/1298
Fixes https://github.com/lovasoa/SQLpage/issues/372
Closes https://github.com/sqlparser-rs/sqlparser-rs/pull/1299

### CODE REVIEW

**Comment by lovasoa** [comment]
@alamb , could you have a look ? The PR is quite simple, it extends the tokenizer to be able to parse [postgresql custom operators](https://www.postgresql.org/docs/current/sql-createoperator.html).

I (and @amrutadotorg) need that for SQLPage.

File: src/tokenizer.rs [hunk 1]
```diff
@@ -1206,6 +1212,41 @@ impl<'a> Tokenizer<'a> {
...
+            }
+            s.push(ch);
+            is_custom_operator = true;
+            chars.next();
+        }
+        if is_custom_operator {
+            Ok(Some(Token::CustomBinaryOperator(s)))
+        } else {
+            Ok(Some(default))
+        }
```
Comment (jmhain) [inline src/tokenizer.rs:1247]: We could avoid the string allocation for the common case of a non-custom binop, something like:

```suggestion
        let mut custom = None;
        while let Some(&ch) = chars.peek() {
            if !self.dialect.is_custom_operator_part(ch) {
                break;
            }

            custom.get_or_insert_with(|| prefix.to_string()).push(ch);
            chars.next();
        }

        Ok(Some(custom.map(Token::CustomBinaryOperator).unwrap_or(default)))
```

**Review by jmhain — APPROVED** [review]
nice, this looks great! cc @alamb

File: src/tokenizer.rs (hunk 1 above)
Comment (lovasoa) [reply src/tokenizer.rs:1247]: Very True ! Let's do it

Comment (lovasoa) [reply src/tokenizer.rs]: done

File: tests/sqlparser_postgres.rs [hunk 2]
```diff
@@ -1770,18 +1793,73 @@ fn parse_pg_binary_ops() {
...
+    }
+}
+
+#[test]
+fn parse_pg_custom_binary_ops() {
+    // Postgres supports declaring custom binary operators, using any character in the following set:
+    //  + - * / < > = ~ ! @ # % ^ & | ` ?
+
+    // Here, we test the ones used by common extensions
+    let operators = [
```
Comment (alamb) [inline tests/sqlparser_postgres.rs:1806]: 😮  -- very cool -- thanks @lovasoa

**Review by alamb — APPROVED** [review]
Thank you @lovasoa  and @jmhain

**Comment by lovasoa** [comment]
Thank you for merging ! This will open a world of new possibilities in SQLPage, using new postgres extensions.

**Pull request merged**